torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -976,6 +978,33 @@ def test_average_parameters(self):
                 for p in model.parameters():
                     self.assertEqual(p.data, torch.ones_like(p.data) * 0.5)
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "MPI backend does not support creating subgroups on CUDA devices",
+        )
+        @skip_if_lt_x_gpu(2)
+        @skip_if_odd_num_of_gpus()
+        def test_periodic_model_averager(self):
+            rank = dist.get_rank()
+            rank_to_GPU = self._init_multigpu_helper()
+            device_id = rank_to_GPU[rank][0]
+            world_size = dist.get_world_size()
+
+            model = nn.Linear(1, 5, bias=False).cuda(device_id)
+            param = next(model.parameters())
+            tensor = torch.ones_like(param.data) * ((rank + 1) // 2)
+            averager = averagers.PeriodicModelAverager(model, warmup_steps=10, period=4)
+            for step in range(0, 20):
+                # Reset the parameters at every step.
+                param.data = copy.deepcopy(tensor)
+                averager.average_parameters(step)
+                if step < 10 or step % 4 == 0:
+                    # After the model averaging is 0.5
+                    self.assertEqual(param.data, torch.ones_like(param.data) * 0.5)
```

Comment:
Does it work for larger groups? 

If world_size = 4, then tensor will be 0.5, 1, 1.5, 2 on respective ranks and the average would be 1.25?

Reply:
Previously I set the value as `(rank + 1) // 2`, so the value will only be 0 or 1.

Anyway, I updated the test to make it easier to understand.